### ISSUE TITLE
Revise visual representations of focus, selection, etc

### ISSUE THREAD

**Issue by danrabbit** [elementary/files#473]
- [ ] Depends on #472 

It's come up that it's a bit unclear and inconsistent how we represent various states between the different views. Attached is a mockup that hopefully clears things up

![bitmap](https://user-images.githubusercontent.com/7277719/39832446-0d3fcffc-537c-11e8-8e55-bb424c18f243.png)

**Comment by jeremypw** [comment]
Yeah, TreeView seems to do it natively whereas IconView doesn't and I am struggling to find the right css to do it on IconView.  The first step was to get the blue rectangle - now I need to get it to turn grey when not keyboard focus.  Notice that in master, it grey all the time and still does not change with keyboard focus.  Both the IconView and TreeView use the same icon and text renderers but the background rectangle is rendered by the widget itself.

**Comment by jeremypw** [comment]
If you want complete consistency between the views, the background rectangle in IconView should turn grey if selected without focus (your issue specified transparent/white)
